app/src/main/java/com/skydoves/pokedex/repository/MainRepository.kt

```diff
@@ -75,5 +77,5 @@ class MainRepository @Inject constructor(
       emit(pokemonDao.getAllPokemonList(page))
       onSuccess()
     }
-  }.onStart { onStart() }.flowOn(Dispatchers.IO)
+  }.onStart { onStart() }.flowOn(Dispatchers.IO).onCompletion { onDone() }
```

Comment:
As I mentioned in the comment, let's change this point using the `onComplete()`.

```kotlin
}.onStart {
    onStart()
  }.onCompletion {
    onComplete()
  }.flowOn(Dispatchers.IO)
```

Reply:
yes. Nicely format